mitosheet/src/types.tsx

```diff
@@ -189,6 +194,45 @@ export type SheetData = {
     columnFormatTypeObjMap: ColumnFormatTypeObjMap
 };
 
+/**
+ * Data about all of the graphs. For each graph, it contains all of the parameters used to construct the graph,
+ * the actual graph html & javascript, and the generated code.
+ * 
+ * @param graphParams - all of the parameters used to construct the graph
+ * @param graphGeneratedCode - the python code to construct the graph
+ * @param graphHTML - the number of rows in the data. Should be equal to data[0].length
+ * @param graphScript - the number of columns in the data. Should be equal to data.length
+ */
+export type GraphData = {
+    graphParams: GraphParams,
+    graphGeneratedCode: string,
+    graphHTML: string | undefined,
+    graphScript: string | undefined,
+};
+
+export type GraphParams = {
+    graphPreprocessing: GraphPreprocessingParams,
+    graphCreation: GraphCreationParams,
+    graphStyling: GraphStylingParams,
+    graphRendering: GraphRenderingParams
+};
+export type GraphPreprocessingParams = {
+    safety_filter_turned_on_by_user: boolean
+}
+export type GraphCreationParams = {
+    graph_type: GraphType,
+    sheet_index: number,
+    x_axis_column_ids: ColumnID[]
+    y_axis_column_ids: ColumnID[]
+}
+export type GraphStylingParams = undefined
+export type GraphRenderingParams = {
+    width?: number
+    height?: number
+}
```

Comment:
Sweet types! Wanna move these above the GraphParams specification just for clarity?